sdk/trace/span_test.go

```diff
@@ -243,3 +244,20 @@ func TestTruncateAttr(t *testing.T) {
 		})
 	}
 }
+
+func BenchmarkRecordingSpanSetAttributes(b *testing.B) {
+	tp := NewTracerProvider(WithSampler(AlwaysSample()))
+	tracer := tp.Tracer("tracer")
+	var attrs []attribute.KeyValue
+	for i := 0; i < 100; i++ {
+		attr := attribute.String(fmt.Sprintf("hello.attrib%d", i), fmt.Sprintf("goodbye.attrib%d", i))
+		attrs = append(attrs, attr)
+	}
+
+	ctx := context.Background()
```

Comment:
How about adding a `b.ResetTimer()` just before the loop?

```suggestion
	ctx := context.Background()
	b.ResetTimer()
```